existing-solutions.tex

```diff
@@ -1,74 +1,71 @@
-there is no downside to staking several branches simultaneously in case of a fork.
+"nothing at stake"\cite{pos_flaws_nothing} problem, which exploits the lack of
+any cost of the actual mining. In this case, there is no downside to staking
+several branches simultaneously in case of a fork.
 
-Ensuring that the source of entropy is distributed along the chain,
-makes all of the elections entirely deterministic and predictable leading to a strategy
-known commonly as "stake grinding," where the dishonest leader tries to rearrange the
+Ensuring that the source of entropy is distributed along the chain, makes all of
+the elections entirely deterministic and predictable leading to a strategy known
+commonly as "stake grinding," where the dishonest leader tries to rearrange the
 transactions to influence the result of the upcoming election.
 
-Next issue is the "long–range attack"\cite{pos_flaws_long}.
-In the very beginning, the stake is scattered among a small
-group of delegates that together have full control over the chain. After
-some time, they can cooperate and start a concurrent chain diverging from
-the main one. This could lead to nasty frauds and would
-destabilize trust over the chain.
+Next issue is the "long–range attack"\cite{pos_flaws_long}. In the very
+beginning, the stake is scattered among a small group of delegates that together
+have full control over the chain. After some time, they can cooperate and start
+a concurrent chain diverging from the main one. This could lead to nasty frauds
+and would destabilize trust over the chain.
 
-On the other hand, there are multiple approaches to deal with these problems. For instance,
-to prevent nothing–at–stake, the CASPER protocol introduces a "wrong voting
-penalty," which punishes
-voters who support conflicting forks\cite{casper}.
-However, this solution is backed by a finality gadget, which is located on another blockchain
-anyway. NXT deals with long–range attack by forcefully
-finalizing all blocks that are older than 720 generations\cite{nxt}.
-One must note that this doesn't actually solve the problem, but rather defers
-it. Moreover, it introduces weak subjectivity since one still needs to trust
-some entity while entering the network for the first time or after a longer
-downtime. Ouroboros staking system has managed to reach solid security, but
-at the cost of very high complexity\cite{ouroboros}.
+On the other hand, there are multiple approaches to deal with these problems.
+For instance, to prevent nothing–at–stake, the CASPER protocol introduces a
+"wrong voting penalty," which punishes voters who support conflicting
+forks\cite{casper}. However, this solution is backed by a finality gadget, which
+is located on another blockchain anyway. NXT deals with long–range attack by
+forcefully finalizing all blocks that are older than 720 generations\cite{nxt}.
+We believe that this doesn not actually solve the problem, but rather defers it.
```

Comment:
```suggestion
We believe that this does not actually solve the problem, but rather defers it.
```

Reply:
Given that Aeternity Mainnet also has forced finality (100 blocks), I would argue that the approach has merit, even if it's not by itself the final answer.